src/system/disposables/DictionaryDisposable.ts

```diff
@@ -0,0 +1,66 @@
+// notice_start
+/*
+ * Copyright 2015 Dev Shop Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// notice_end
+
+import {DisposableWrapper} from './DisposableWrapper';
+import {Disposable} from './disposable';
+
+export class DictionaryDisposable implements Disposable {
+    private _disposables: Map<string, Disposable>;
+    private _isDisposed: boolean;
+
+    public constructor() {
+        this._isDisposed = false;
+    }
+
+    public get isDisposed(): boolean {
+        return this._isDisposed;
+    }
+
+    public add(key: string, disposable: Disposable) {
+        if (this._disposables.has(key)) {
+            throw new Error("Key " + key + " already found");
+        }
+        let disposableWrapper = new DisposableWrapper(disposable);
+        if (this._isDisposed) {
+            disposableWrapper.dispose();
+            return;
+        }
+        this._disposables.set(key, disposableWrapper);
+    }
+
+    public remove(key): boolean {
+        if (this._disposables.has(key)) {
+            this._disposables.delete(key);
+            return true;
+        }
+        return false;
+    }
+
+    public containsKey(key): boolean {
+        return this._disposables.has(key);
+    }
+
+    public dispose(): void {
+        this._isDisposed = true;
+        this._disposables.forEach(disposable => {
+            if (disposable.dispose) {
```

Comment:
No longer have to do this check.  The disposables are all guaranteed to be a `DisposableWrapper` which will ahve a dispose function.

Reply:
it's I guess possible that something trashed the method by the time this is called. Unlikely, but that's why that code existed.